styleguide/drafts/tooling.rst

```diff
@@ -0,0 +1,518 @@
+Tooling
+=======
+This Section focuses on development tools we are using for our python based projects.
+If you have a specific need to address (code formatting, linting, etc.) you should use
+one of our "standard" tools in this list. If your need(s) aren't covered by the tools
+listed here, we recommend you propose an addition to this list.
+
+Justification
+_____________
+Unifying basic tooling removes overall complexity when dealing with multiple projects.
+
+* Reduces on-boarding of new people into the project(s)
+* Reduces cognitive overhead when switching between projects
+
+  * Running Test, Build is the same in "every"  project e.g. `nox -s test`
+  * Developers are more likely to develop deep knowledge of the tools
+
+* Reduces the potential of errors and misuse
+* Increases potential for automation (checks, automated fixes, etc.)
+* Decreases complexity of automation
+
+Overview
+________
+
+Task Runner & Automation
+++++++++++++++++++++++++
+
+* Nox_       (task runner, build automation, test & build isolation)
+* PreCommit_ (commit hooks)
+
+Code Style
+++++++++++
+
+* Black_     (style, code formatter)
+* Isort_     (style, import sorting)
+
+Testing
++++++++
+
+* Pytest_    (ut, testing)
+
+Documentation
++++++++++++++
+
+* Sphinx_    (documentation)
+
+Dependencies & Packaging
+++++++++++++++++++++++++
+
+* Poetry_    (dependency management, virtual environment, build-tool, packaging)
+
+Code Health
++++++++++++
+
+* Pylint_    (linter, code analysis)
+* Coverage_  (coverage)
+* MyPy_      (linter, type checking)
+* Sonar_     (linter, code coverage, code health)
```

Comment:
what do you think, will pylin and sonar collide?

Reply:
Sonar supports pylint reports see https://docs.sonarcloud.io/enriching/external-analyzer-reports/